ansible_base/resource_registry/models/resource.py

```diff
@@ -40,6 +41,21 @@ def can_be_managed(self):
     def get_resource_config(self):
         return self.resource_registry.get_config_for_model(model=ContentType.objects.get_for_id(self.content_type_id).model_class())
 
+    def get_conflicting_resource(self, resource_data):
+        filter = {}
+
+        serialized_data = self.serializer_class(data=resource_data)
+        serialized_data.is_valid(raise_exception=True)
+        serialized_data = serialized_data.validated_data
```

Comment:
Do we ever have to consider a case in here where the data isn't valid like the ones with the user IDs and wrap this in a try/catch to prevent an exception?

Reply:
I think I prefer to let the caller catch this one. I'm not sure what we'd do if it was not valid. We'd just have to raise another exception.